WordPress/Classes/ViewRelated/Blog/Blog Dashboard/Cards/Posts/DashboardPostsCardCell.swift

```diff
@@ -29,60 +32,62 @@ class DashboardPostsCardCell: UICollectionViewCell, Reusable, BlogDashboardCardC
             return
         }
 
+        self.viewController = viewController
+        self.blog = blog
+
         let hasDrafts = (apiResponse.posts?.draft?.count ?? 0) > 0
         let hasScheduled = (apiResponse.posts?.scheduled?.count ?? 0) > 0
+        let hasPublished = apiResponse.posts?.hasPublished ?? true
 
         removeAllChildVCs()
 
         if !hasDrafts && !hasScheduled {
-            let hasPublished = apiResponse.posts?.hasPublished ?? true
-            // Temporary: it should display "write your next post"
-            let postsViewController = PostsCardViewController(blog: blog, status: .draft)
-            draftPostsViewController = postsViewController
-
-            let cardTitle = hasPublished ? Strings.nextPostTitle : Strings.firstPostTitle
-            embed(child: postsViewController, to: viewController, with: cardTitle)
+            showCard(forBlog: blog, status: .draft, to: viewController, hasPublishedPosts: hasPublished)
         } else {
             if hasDrafts {
-                let postsViewController = PostsCardViewController(blog: blog, status: .draft)
-                draftPostsViewController = postsViewController
-
-                embed(child: postsViewController, to: viewController, with: Strings.draftsTitle)
+                showCard(forBlog: blog, status: .draft, to: viewController, hasPublishedPosts: hasPublished)
             }
 
             if hasScheduled {
-                let postsViewController = PostsCardViewController(blog: blog, status: .scheduled)
-                scheduledPostsViewController = postsViewController
-
-                embed(child: postsViewController, to: viewController, with: Strings.scheduledTitle)
+                showCard(forBlog: blog, status: .scheduled, to: viewController, hasPublishedPosts: hasPublished)
             }
         }
     }
 
     private func removeAllChildVCs() {
-        stackView.removeAllSubviews()
-
-        if let draftPostsViewController = draftPostsViewController {
-            remove(child: draftPostsViewController)
-        }
+        let childVcs = viewController?.children.filter { $0 is PostsCardViewController }
 
-        if let schedulePostsViewController = scheduledPostsViewController {
-            remove(child: schedulePostsViewController)
-        }
+        stackView.removeAllSubviews()
 
-        draftPostsViewController = nil
-        scheduledPostsViewController = nil
+        childVcs?.forEach { remove(child: $0) }
     }
 
-    private func embed(child childViewController: UIViewController, to viewController: UIViewController, with title: String) {
+    private func showCard(forBlog blog: Blog, status: BasePost.Status, to viewController: UIViewController, hasPublishedPosts: Bool, hiddenHeader: Bool = false) {
```

Comment:
[nit]
```suggestion
    private func showCardForBlog(_ blog: Blog, status: BasePost.Status, in viewController: UIViewController, hasPublishedPosts: Bool, hidesHeader: Bool = false) {
```

Reply:
I'll address this in an upcoming PR that changes this function!